src/dbnode/storage/bootstrap/bootstrapper/commitlog/source_data_test.go

```diff
@@ -384,10 +433,13 @@ func TestItMergesSnapshotsAndCommitLogs(t *testing.T) {
 	expectedValues := append([]testValue{}, snapshotValues...)
 	expectedValues = append(expectedValues, commitLogValues[0:3]...)
 
-	require.NoError(t, verifyShardResultsAreCorrect(
-		expectedValues, blockSize, res.ShardResults(), opts))
+	require.NoError(t, verifyShardResultsAreCorrect(nsCtx,
+		expectedValues, blockSize, res.ShardResults(), opts, annEqual))
 }
 
+type setAnnotation func([]testValue) []testValue
+type annotationEqual func([]byte, []byte) bool
```

Comment:
Hm this is definitely a little strange.  Can't we infer whether the namespace is using proto or not and just use the correctly annotation equal fn?

Reply:
we could, though it might be cleaner if we don't infer. I image we could infer by whether or not schema is nil in nsCtx, though if nsCtx is not set correctly, the test would have succeed anyway even not using the right comparator. so not infer, always explicit seem to be better.